### PR TITLE
[#317] Use vega for charts

### PR DESCRIPTION
**Pull request by gabemart**
- [ ] Test plan | Unit test | Integration test
- [ ] Copyright header
- [ ] Code formatting
- [ ] Documentation

### CODE REVIEW

**Comment by jonase** [comment]
@gabemart 

When I `npm install` this branch I see the following error

```
▶ npm install

> canvas@1.4.0 install /Users/jonasenlund/dev/akvo/akvo-lumen/client/node_modules/canvas
> node-gyp rebuild

Package cairo was not found in the pkg-config search path.
Perhaps you should add the directory containing `cairo.pc'
to the PKG_CONFIG_PATH environment variable
No package 'cairo' found
gyp: Call to './util/has_lib.sh freetype' returned exit status 0 while in binding.gyp. while trying to load binding.gyp
gyp ERR! configure error
gyp ERR! stack Error: `gyp` failed with exit code: 1
gyp ERR! stack     at ChildProcess.onCpExit (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/configure.js:305:16)
gyp ERR! stack     at emitTwo (events.js:106:13)
gyp ERR! stack     at ChildProcess.emit (events.js:191:7)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:204:12)
gyp ERR! System Darwin 15.6.0
gyp ERR! command "/usr/local/Cellar/node/4.1.1/bin/node" "/usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /Users/jonasenlund/dev/akvo/akvo-lumen/client/node_modules/canvas
gyp ERR! node -v v6.2.2
gyp ERR! node-gyp -v v3.3.1
gyp ERR! not ok
npm WARN install:canvas@1.4.0 canvas@1.4.0 install: `node-gyp rebuild`
npm WARN install:canvas@1.4.0 Exit status 1
```

Have you seen this too?

**Comment by gabemart** [comment]
No, I'll try nuking node_modules and reinstalling

File: client/package.json [hunk 1]
```diff
@@ -59,6 +59,7 @@
     "style-loader": "^0.13.1",
     "tus-js-client": "^1.1.4",
     "url-loader": "^0.5.7",
+    "vega": "^2.6.3",
```
Comment (jonase) [inline client/package.json]: remove `rd3` dependency?

File: client/src/containers/Visualisation.jsx [hunk 2]
```diff
@@ -177,16 +177,17 @@ class Visualisation extends Component {
       return <div className="Visualisation">Loading...</div>;
     }
     const { VisualisationHeader, VisualisationEditor } = this.state.asyncComponents;
+    const clonedVisualisation = cloneDeep(this.state.visualisation);
```
Comment (jonase) [inline client/src/containers/Visualisation.jsx]: Is cloning necessary here? A quick look at where we update `this.state.visualisation` via `setState` we seem to try to do it functionally via `Object.assign` or `update`.

**Comment by jonase** [comment]
👍 lgtm

**Pull request merged**